src/App.test.tsx

```
@@ -0,0 +1,255 @@
+import { vi } from 'vitest';
+import { router as appRouter } from '@/App';
+import { ThemeContext } from '@/context/ThemeContext';
+import { HabitsStateProvider } from '@/features/habits/context/HabitsStateContext';
+// Do not import useHabits here if its module is being fully mocked with a factory
+// that references variables from this file that might not be hoisted correctly.
+// We will import the mocked version later inside describe/beforeEach.
+import { Habit } from '@/features/habits/types/Habit';
+
+// Mock framer-motion
+vi.mock('framer-motion', async (importOriginal) => {
+  const actual = await importOriginal() as any; // Use 'as any' for simplicity with importOriginal
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)